winnery_ticket_experiment.py

```diff
@@ -60,20 +61,25 @@ def get_updated_weights_by_mask(self, layer_weights, zero_weights_mask):
     def go(self):
         super(WinneryTicketExperiment, self).go()
 
+        base_trainer = self.get_trainer_map()[self.base_model_key]
+        layer_indices = base_trainer.get_weighted_layers_indices()
+
         with self.open_model(self.base_model_key) as trained_model:
-            prunning_mask = BaseTrainer.prune_trained_model(trained_model, self.prune_threshold)
+            prunning_mask = BaseTrainer.prune_model(trained_model, self.prune_threshold, layer_indices)
 
         # Setup a new trainer with weights "pruned".
-        init_model = self._get_model_at_epoch(self.base_model_key, 'start')
-        weight_map = {i: self.get_updated_weights_by_mask(init_model.layers[i].get_weights(),
-                                                          prunning_mask[i - 1])
-                      for i in range(1, len(init_model.layers))}
+        with self.open_model_at_epoch(self.base_model_key, 'start') as init_model:
+            weight_map = {layer: self.get_updated_weights_by_mask(init_model.layers[layer].get_weights(),
+                                                              prunning_mask[i])
+                          for i, layer in enumerate(layer_indices)}
```

Comment:
`pruning_mask[0]` is the mask for `init_model.layers[1]`? If so then OK

Reply:
Yes, we count layers from 1..L and the mask is 0...L-1 :)